src/silx/gui/plot/ImageStack.py

```diff
@@ -145,49 +147,90 @@ def _urlChanged(self, value):
         self.sigCurrentUrlIndexChanged.emit(value)
 
 
-class UrlList(qt.QWidget):
+class UrlList(qt.QListWidget):
     """List of URLs the user to select an URL"""
 
     sigCurrentUrlChanged = qt.Signal(str)
-    """Signal emitted when the active/current url change"""
+    """Signal emitted when the active/current url change. If url == '' consider it as there is not data to be displayed"""
+
+    sigUrlsRemoved = qt.Signal(tuple)
```

Comment:
What about having the signal triggered for each removed URL?

```suggestion
    sigUrlRemoved = qt.Signal(str)
```

Reply:
This won't change much from my side. We cam go for it if you think it is more convenient...

Reply:
No problem if you think this is more convenient

Reply:
I think it's more consistent with other Qt signals